### ISSUE TITLE
Dynamic rule filtering does not propagate to destination subdomains for `sub_frame` type

### ISSUE THREAD

**Issue by farnoy** [uBlockOrigin/uBlock-issues#2157]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

The documentation [states](https://github.com/uBlockOrigin/uBlock-issues/wiki/Dynamic-filtering:-rule-syntax):

> A rule always automatically propagates to all subdomains of the source hostname and all subdomains of the destination hostname -- unless the rule is overridden by a narrower rule in one of the subdomains.

But either this is not true for the `sub_frame` type, or somehow a more generic rule takes precedence over this more specific one.

### A specific URL where the issue occurs

docs.google.com/spreadsheets

### Steps to Reproduce

1. add a dynamic uBO rule `* * 3p-frame block`
2. Add a second rule `google.com https://googleusercontent.com sub_frame noop`
3. Open a Google Docs Spreadsheet
4. Try to print the spreadsheet
5. 

### Expected behavior

The download should have worked, as the request happens from `docs.google.com` and the target domain is `https://doc-04-9s-sheets.googleusercontent.com` in my case.

A more specific rule like `google.com https://doc-04-9s-sheets.googleusercontent.com sub_frame  noop` works perfectly fine, that's why I suspect subdomain propagation to be the issue. I don't want to use this specific filter because the domains they use for downloads vary.

### Actual behavior

uBO blocks the download file action that GDocs runs in an iframe, the inspector shows the reason is the global `* * 3p-frame block` rule

### uBlock Origin version

1.43.0

### Browser name and version

FF developer edition 102.0b9

### Operating System and version

Windows 11

**Comment by gwarser** [comment]
You are mixing dynamic filtering with URL filtering.

https://github.com/uBlockOrigin/uBlock-issues/wiki/Dynamic-URL-filtering

>destination URL is URL to match: the URL of a resource must start exactly with the destination URL of a rule for a match to occur.

**Comment by farnoy** [comment]
I see, so the ideal solution to override the global dynamic rule `* * 3p-frame block` would be to use a dynamic local rule like `google.com googleusercontent.com 3p-frame noop`, but dynamic local rules don't support specific request types.

The closest approximation I've found is `google.com googleusercontent.com * noop`, which works but is imprecise compared to what I intended.

That's good enough I guess, thanks for pointing this out.
